Classes/Finisher/RedirectFinisher.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+namespace FriendsOfTYPO3\HeadlessPowermail\Finisher;
+
+use In2code\Powermail\Domain\Service\RedirectUriService;
+use In2code\Powermail\Utility\FrontendUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+class RedirectFinisher extends \In2code\Powermail\Finisher\RedirectFinisher
+{
+    /**
+     * Redirect user after form submit
+     *
+     * @return void
+     */
+    public function redirectToUriFinisher(): void
+    {
+        $redirectService = GeneralUtility::makeInstance(RedirectUriService::class, $this->contentObject);
+        $uri = $redirectService->getRedirectUri();
+        if (!empty($uri) && $this->isRedirectEnabled()) {
+            echo json_encode([
+                'redirectUrl' => $uri,
+                'statusCode' => 303,
+            ]);
+            die;
```

Comment:
I don't think it's proper way to finish

Reply:
@lukaszuznanski, that's how "the core" did it ;-)
https://github.com/TYPO3/typo3/blob/11.5/typo3/sysext/core/Classes/Utility/HttpUtility.php#L110

Reply:
fair enough ;)